api/v1alpha8/openstackcluster_types.go

```diff
@@ -167,6 +167,8 @@ type OpenStackClusterSpec struct {
 
 // OpenStackClusterStatus defines the observed state of OpenStackCluster.
 type OpenStackClusterStatus struct {
+	// Ready is true when the cluster infrastructure is ready.
+	// +optional
```

Comment:
well this is in fact wrong:

https://cluster-api.sigs.k8s.io/developer/providers/cluster-infrastructure#data-types

I'll investigate

Reply:
* GCP is setting it as required: https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/api/v1beta1/gcpcluster_types.go#L75
* AWS: default to false: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1beta2/awscluster_types.go#L252
* Azure: optional: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azurecluster_types.go#L66C1-L66C14

The GCP is doing correctly but I suspect it works because they never patch an empty cluster object while in our case we patch empty cluster when the bastion instance is still being built.

Reply:
@mdbooth when time permits in a reasonable future we need to get back to this.

Reply:
This may be a stupid question, but does it help if we set `omitempty`? My understanding would be that no value would then be interpreted as `Ready: false`

Reply:
I also expected no value to equals false, however this does not seem to be the case. I stumbeled upon this issue and it seems to break the chagnes implemented by https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1829

To be honest I think the AWS "solution" looks most appealing. However does not really have an optionion if we could get this resolved.

Currently this however causing Floating IPS to leak and casuing the problem mentioned in https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/1687